diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 485499d..35c95fd 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -5,11 +5,6 @@
 ###################################################################################################
 
 CMAKE_MINIMUM_REQUIRED(VERSION 3.16)
-
-SET(X_VCPKG_APPLOCAL_DEPS_INSTALL ON)		# Install vcpkg dependencies automatically (experimental - might be changed or removed later; see: https://github.com/microsoft/vcpkg/issues/1653). 
-SET(VCPKG_MODULE_DIRECTORY "${CMAKE_SOURCE_DIR}/modules/vcpkg" CACHE STRING "VCPKG module directory." FORCE)
-SET(CMAKE_TOOLCHAIN_FILE "${VCPKG_MODULE_DIRECTORY}/scripts/buildsystems/vcpkg.cmake" CACHE STRING "VCPKG toolchain file." FORCE)
-
 PROJECT(LiteFX LANGUAGES CXX)
 
 # Check if the build dir does not match the source dir (disallow in-source builds).
diff --git a/src/Core/CMakeLists.txt b/src/Core/CMakeLists.txt
index 6a71fdc..392f823 100644
--- a/src/Core/CMakeLists.txt
+++ b/src/Core/CMakeLists.txt
@@ -10,9 +10,9 @@ MESSAGE(STATUS "Initializing: ${PROJECT_NAME}...")
 # Resolve package dependencies.
 FIND_PACKAGE(fmt CONFIG REQUIRED)
 
-CONFIGURE_FILE("config.tmpl" "${CMAKE_CURRENT_SOURCE_DIR}/include/litefx/config.h")
-CONFIGURE_FILE("version.tmpl" "${CMAKE_CURRENT_SOURCE_DIR}/include/litefx/version.h")
-CONFIGURE_FILE("core.tmpl" "${CMAKE_CURRENT_SOURCE_DIR}/include/litefx/core.h")
+CONFIGURE_FILE("config.tmpl" "${CMAKE_CURRENT_BINARY_DIR}/include/litefx/config.h")
+CONFIGURE_FILE("version.tmpl" "${CMAKE_CURRENT_BINARY_DIR}/include/litefx/version.h")
+CONFIGURE_FILE("core.tmpl" "${CMAKE_CURRENT_BINARY_DIR}/include/litefx/core.h")
 
 # Collect header & source files.
 SET(CORE_HEADERS
@@ -20,9 +20,6 @@ SET(CORE_HEADERS
     "include/litefx/string.hpp"
     "include/litefx/traits.hpp"
     "include/litefx/exceptions.hpp"
-    "include/litefx/core.h"
-    "include/litefx/config.h"
-    "include/litefx/version.h"
     "include/litefx/litefx.h"
 )
 
@@ -33,6 +30,9 @@ SET(CORE_SOURCES
 ADD_LIBRARY(${PROJECT_NAME} SHARED
     ${CORE_HEADERS}
     ${CORE_SOURCES}
+    "${CMAKE_CURRENT_BINARY_DIR}/include/litefx/config.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/include/litefx/version.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/include/litefx/core.h"
 )
 
 # Create source groups for better code organization.
@@ -59,9 +59,13 @@ TARGET_INCLUDE_DIRECTORIES(${PROJECT_NAME}
         "src/"
     PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
         $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include>
 )
 
+# Ignore codecvt deprication warnings (as long as there is no replacement, the header will continue to work).
+TARGET_COMPILE_DEFINITIONS(${PROJECT_NAME} PUBLIC _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING)
+
 # Link project dependencies.
 TARGET_LINK_LIBRARIES(${PROJECT_NAME} PUBLIC fmt::fmt)
 
diff --git a/src/Core/core.tmpl b/src/Core/core.tmpl
index 5d61e94..87757d2 100644
--- a/src/Core/core.tmpl
+++ b/src/Core/core.tmpl
@@ -8,5 +8,5 @@
 #  include <Windows.h>
 #endif
 
-#include "containers.hpp"
-#include "traits.hpp"
\ No newline at end of file
+#include <litefx/containers.hpp>
+#include <litefx/traits.hpp>
\ No newline at end of file
diff --git a/src/Math/CMakeLists.txt b/src/Math/CMakeLists.txt
index 6350404..b2f50cd 100644
--- a/src/Math/CMakeLists.txt
+++ b/src/Math/CMakeLists.txt
@@ -7,12 +7,17 @@
 PROJECT(LiteFX.Math VERSION ${LITEFX_VERSION} LANGUAGES CXX)
 MESSAGE(STATUS "Initializing: ${PROJECT_NAME}...")
 
+# Find optional dependencies.
+SET(DEPENDENCY_PACKAGES "")
+
 IF(BUILD_WITH_GLM)
     FIND_PACKAGE(glm CONFIG REQUIRED)
+    LIST(APPEND DEPENDENCY_PACKAGES "glm::glm")
 ENDIF(BUILD_WITH_GLM)
 
 IF(BUILD_WITH_DIRECTX_MATH)
     FIND_PACKAGE(directxmath CONFIG REQUIRED)
+    LIST(APPEND DEPENDENCY_PACKAGES "Microsoft::DirectXMath")
 ENDIF(BUILD_WITH_DIRECTX_MATH)
 
 # Collect header & source files.
@@ -58,7 +63,7 @@ TARGET_INCLUDE_DIRECTORIES(${PROJECT_NAME}
 
 # Link project dependencies.
 TARGET_LINK_LIBRARIES(${PROJECT_NAME}
-    PUBLIC LiteFX.Core $<$<BOOL:BUILD_WITH_GLM>:glm::glm> $<$<BOOL:BUILD_WITH_DIRECTX_MATH>:Microsoft::DirectXMath> 
+    PUBLIC LiteFX.Core ${DEPENDENCY_PACKAGES}
 )
 
 # Re-use pre-compiled core header.
diff --git a/src/cmake/Config.cmake b/src/cmake/Config.cmake
index 075eb4a..56cc4f1 100644
--- a/src/cmake/Config.cmake
+++ b/src/cmake/Config.cmake
@@ -24,11 +24,10 @@ ENDIF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 SET(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/binaries/")
 
 # Setup installation directories.
-SET (CMAKE_INSTALL_LIBRARY_DIR "lib")
-SET (CMAKE_INSTALL_BINARY_DIR  "bin")
+SET (CMAKE_INSTALL_LIBRARY_DIR "${CMAKE_INSTALL_LIBDIR}")
+SET (CMAKE_INSTALL_BINARY_DIR  "${CMAKE_INSTALL_BINDIR}")
+SET (CMAKE_INSTALL_INCLUDE_DIR "${CMAKE_INSTALL_INCLUDEDIR}")
 SET (CMAKE_INSTALL_EXPORT_DIR  "cmake")
-SET (CMAKE_INSTALL_INCLUDE_DIR "include")
-SET (CMAKE_INSTALL_INCLUDE_DIR "include")
 
 # Export all symbols automatically on Windows.
 IF (WIN32)
